lib/ecto_network/inet.ex

```diff
@@ -8,14 +8,15 @@ defmodule EctoNetwork.INET do
   def type, do: :inet
 
   @doc "Handle casting to Postgrex.INET"
-  def cast(%Postgrex.INET{}=address), do: {:ok, address}
-  def cast(address) when is_binary(address) do
-    case parse_address(address) do
-      {:ok, parsed_address} -> {:ok, %Postgrex.INET{address: parsed_address}}
-      {:error, _einval}     -> :error
+  def cast(address) do
+    address
+    |> case do
+      %Postgrex.INET{}=address -> {:ok, address}
```

Comment:
I consolidated the functions and refactored to rely upon recursion. The casting should always eventually result in a `Postgrex.INET` or `:error`